src/Orchard/FileSystems/Media/FileSystemStorageProvider.cs

```diff
@@ -427,6 +458,16 @@ public class FileSystemStorageProvider : IStorageProvider {
             return (di.Attributes & FileAttributes.Hidden) != 0;
         }
 
+        public static bool FolderNameContainsInvalidCharacters(string folderName, out char[] invalidCharacters) {
+            invalidCharacters = folderName.Distinct().Intersect(InvalidFolderNameCharacters).ToArray();
+            return invalidCharacters.Any();
+        }
+
+        public static bool FileNameContainsInvalidCharacters(string folderName, out char[] invalidCharacters) {
+            invalidCharacters = folderName.Distinct().Intersect(InvalidFileNameCharacters).ToArray();
```

Comment:
string.IndexOfAny()

Reply:
Addressed in 15dc19ce9c9599496b9502c2666d4817b37c1ff1.